src/utils.c

```diff
@@ -160,7 +160,7 @@ void coerceFill(SEXP fill, double *dfill, int32_t *ifill, int64_t *i64fill) {
         // NA -> NA, NaN -> NaN
         ifill[0] = NA_INTEGER; dfill[0] = rfill; i64fill[0] = NA_INTEGER64;
       } else {
-        ifill[0] = rfill>INT32_MAX ? NA_INTEGER : (int32_t)(rfill);
+        ifill[0] = rfill>INT32_MAX || !R_finite(rfill) ? NA_INTEGER : (int32_t)(rfill);
```

Comment:
I think we used capital case R_FINITE, should be fine to use it here too, you can grep for both and double check

Reply:
I didn't know `R_finite` was available actually. It seems to compile ok by the looks of it.   R 2.6.0 news contains this :
![image](https://user-images.githubusercontent.com/2779998/66172304-4136a700-e600-11e9-937f-18569b9e06cb.png)
However, section 6.4 of R-exts points to the macro form (capitals) : 
https://cran.r-project.org/doc/manuals/r-devel/R-exts.html#Missing-and-IEEE-values
I can't see any mention of `R_finite` in R-exts, only `R_FINITE`.  So that's the reason to use `R_FINITE`. Also R_FINITE should be tested first in the `||` to use the short-circuit and avoid doubt over `rfill>INT32_MAX` on non-finite values.

Reply:
What about `<= INT32_MIN` ?     ( `<=` rather than `<` due to NA_INTEGER )

Reply:
Is `<=INT_MIN` needed on line 153 too?  https://github.com/Rdatatable/data.table/pull/3930/files#diff-74710e3ecbe94109f39de34133e1c8caR153